src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -367,4 +369,88 @@ public boolean contains(String key) {
 
         return mPrefs.contains(key);
     }
+    
+    public TokenCacheItem getFamilyRefreshTokenItemForUser(String userId) {
+        Iterator<TokenCacheItem> results = this.getAll();
+        TokenCacheItem tokenItem = new TokenCacheItem();
+        while (results.hasNext()) {
+            final TokenCacheItem tokenCacheItem = results.next();
+            if(tokenCacheItem.getUserInfo() != null
+                    && tokenCacheItem.getUserInfo().getUserId().equalsIgnoreCase(userId)
+                    && !StringExtensions.IsNullOrBlank(tokenCacheItem.getFamilyClientId())
+                    && tokenCacheItem.getFamilyClientId().equalsIgnoreCase("1")){
+                tokenItem = tokenCacheItem;
+                //return the cacheItem which contains the FRT for this userId
+                //there should be at most only one FRT record for each userId
+                return tokenItem;
+            }
+        }
+        //If the cache does not contain the FRT for this user return null
+        return null;
+    }
+    
+    public String serialize(String uniqueUserId) throws AuthenticationException{
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if((new BrokerProxy(mContext)).canSwitchToBroker()){
+            throw new AuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
```

Comment:
Thank you for your reviews, but mBrokerProxy is private in AuthenticationContext and there is no getter for it.